src/Context/ProductsContext.js

```diff
@@ -20,13 +28,13 @@ const ProductProvider = ({children}) => {
     useEffect(() => {
         
         (async() => {
-            productsDispatch({type: ACTION_TYPE_LOADING})
+            productsDispatch({type: "ACTION_TYPE_LOADING"})
             try{         
                 const response = await axios.get("/api/products");
-                productsDispatch({type: ACTION_TYPE_SUCCESS, payload: response.data.products})
+                productsDispatch({type: "ACTION_TYPE_SUCCESS", payload: response.data.products})
             }
             catch{
-            productsDispatch({type: ACTION_TYPE_ERROR, payload:"Error in fetching the product data"})
+            productsDispatch({type: "ACTION_TYPE_ERROR", payload:"Error in fetching the product data"})
             }
         })()
```

Comment:
Just a suggestion you can create a custom hook for fetching data.

Reply:
yes will surely implement a custom hook to fetch the data from backend.